http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -354,17 +354,33 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         F.blocking(wsmApi.getAzureManagedIdentity(workspaceId.value, wsmIdentity.resourceId.value))
       }
 
+      // Create or fetch WSM managed identity if shared app
```

Comment:
It looks like this code is enumerating all managed identities for the workspace in WSM, and choosing the right one. Are lines 350-355 needed in this case? Previously we would store the WSM IDs of an app's constituent resources in the `APP_CONTROLLED_RESOURCE` table, so Leo could retrieve the specific WSM records in the `updateApp` and `deleteApp` flows. Are we continuing to use `APP_CONTROLLED_RESOURCE`, or just use WSM as the only source of truth?

Reply:
I actually think it would be cleaner and more resilient if we did use WSM as the source of truth instead of going into contortions to keep a local Leo table up to date with WSM tables, when we do not get notification from WSM when tables are created.

Reply:
OK - yeah if we use WSM as the source of truth we should do it consistently for `updateApp` and `deleteApp`, and ideally drop the `APP_CONTROLLED_RESOURCE` tables.

(That might also align with the changes @aherbst-broad is thinking about to add an API alternative to WSM `enumerateResources`.)